Assets/00_Content/Scripts/Vikings/States/BrawlingVikingState.cs

```diff
@@ -9,39 +11,73 @@ namespace Vikings.States {
 	/// </summary>
 	public class BrawlingVikingState : VikingState {
 		private Table targetTable;
+		private Viking vikingTarget;
+		private PlayerComponent playerTarget;
 		private NavMeshAgent navMeshAgent;
+		private BrawlType brawlType;
 
 		private float attackTimer;
 
 		private bool IsMoving => navMeshAgent.desiredVelocity.sqrMagnitude != 0;
 
 		public BrawlingVikingState(Viking viking, Table targetTable) : base(viking) {
-			Debug.Assert(targetTable != null, "Viking is entering a brawl with no target");
+			Debug.Assert(targetTable != null, "Viking is entering a tableBrawl with no target");
 			this.targetTable = targetTable;
+			brawlType = BrawlType.TableBrawl;
+		}
+
+		public BrawlingVikingState(Viking viking, Viking vikingTarget) : base(viking) {
+			Debug.Assert(vikingTarget != null, "Viking is entering a vikingBrawl with no target");
+			this.vikingTarget = vikingTarget;
+			brawlType = BrawlType.VikingBrawl;
+		}
+
+		public BrawlingVikingState(Viking viking, PlayerComponent playerTarget) : base(viking) {
+			Debug.Assert(playerTarget != null, "Viking is entering a playerBrawl with no target");
+			this.playerTarget = playerTarget;
+			viking.IsSeated = false;
+			brawlType = BrawlType.PlayerBrawl;
 		}
 
 		public override VikingState Enter() {
 			navMeshAgent = viking.GetComponent<NavMeshAgent>();
 			navMeshAgent.enabled = true;
 
 			viking.bodyMeshRenderer.material = viking.brawlingMaterial;
-			viking.DismountChair();
+
+			if(brawlType == BrawlType.TableBrawl)
+				viking.DismountChair();
```

Comment:
How come you're only dismounting the chair if we're attacking a table?

Reply:
Good point, will just add a check to see if the viking is seated before going into a brawlState.